org.eclipse.lsp4j.jsonrpc/src/test/java/org/eclipse/lsp4j/jsonrpc/test/TracingMessageConsumerTest.java

```diff
@@ -299,6 +299,72 @@ public void testSendingNotificationWithCustomAdapter() {
 
 		assertEquals(expectedTrace, actualTrace);
 	}
+
+	@Test
+	public void testUnmatchedResponseWhenOutgoingIsNotStreamConsumer() {
+		// Simulate an outgoing transport that is not StreamMessageConsumer (e.g., websocket)
+		var outgoingTrace = new StringWriter();
+		var outgoingPrintWriter = new PrintWriter(outgoingTrace);
+
+		var incomingTrace = new StringWriter();
+		var incomingPrintWriter = new PrintWriter(incomingTrace);
+
+		var sentRequests = new HashMap<String, RequestMetadata>();
+		var receivedRequests = new HashMap<String, RequestMetadata>();
+
+		// Wrap a non-stream consumer so TracingMessageConsumer does not treat it as "sending"
+		var outgoingTracing = new TracingMessageConsumer(new EmptyMessageConsumer(), sentRequests, receivedRequests, outgoingPrintWriter, TEST_CLOCK_1, Locale.US);
+
+		// Send a request (this will NOT populate sentRequests because consumer is not StreamMessageConsumer)
```

Comment:
This comment also describes the old implementation rather than the new one. I'd just remove it, as it can be misleading.